crates/next-core/src/next_manifests/client_reference_manifest.rs

```diff
@@ -93,6 +96,8 @@ impl ClientReferenceManifest {
             let node_root_ref = &*node_root.await?;
             let rsc_app_entry_chunks = &*rsc_app_entry_chunks.await?;
 
+            let async_module_info = async_module_info.await?;
```

Comment:
Isolate the invalidation, by adding a intermediate turbo-tasks function.

Either `async_module_info.get(...).await?` to get is_async for a single module.
Or `async_module_info.get_all(Vc::cell(...)).await?` to get is_async for all modules at once.